malloc.c

```diff
@@ -651,6 +651,10 @@ GC_API void GC_CALL GC_free(void * p)
         return;
     }
 
+#   ifdef VALGRIND_TRACKING
```

Comment:
I think it would be better to remove "ifdef VALGRIND_TRACKING" everywhere but defining GC_free_profiler_hook(p) macro as (void)(p) in gc_priv.h if VALGRIND_TRACKING is not defined.

Reply:
I've probably misunderstood this. Do you want me to replace the `GC_free_profiler_hook` function with a macro definition? I don't think this would work anymore as there would be no symbol for profilers to intercept.

Reply:
No, I will do it, in some other way.